core/consensus/babe/src/lib.rs

```diff
@@ -769,6 +777,41 @@ fn register_babe_inherent_data_provider(
 	}
 }
 
+/// Register uncles inherent data provider, if not registered already.
+fn register_uncles_inherent_data_provider<B, C, SC>(
```

Comment:
putting this in the BABE crate doesn't seem right - other block production systems will also want uncles